cpp/Test.cpp

```diff
@@ -1,72 +1,80 @@
 #include "Input.h"
 #include "Model.h"
 #include <opencv2/opencv.hpp>
+#include <assert.h>
 
-#define TILES_DIR "../../../tiles/spirals"
+//#define TILES_DIR "../../../tiles/spirals"
 //#define SHOWPATTS
 
-int main() {
+int main(int argc, char** argv) {
+	assert(argc == 4);
+	char* tiles_dir = argv[1];
+	int set = atoi(argv[2]); // should be 0 if single image, 1 if it is a set of tiles
+	int render = atoi(argv[3]); // should be 0 if we don't want to render, 1 if we do
+
+
 	std::vector<Pair> overlays;
 	generate_neighbor_overlay(overlays);
-	Model model(TILES_DIR, Pair(64, 64), 3, overlays, true, true, -1);
+	Model model(tiles_dir, Pair(64, 64), 3, overlays, true, true, -1);
 	
-#ifdef SHOWPATTS
-	auto break_all = false;
-	std::cout << std::endl;
+	if(render) {
+		auto break_all = false;
+		std::cout << std::endl;
 
-	for (auto pat_idx1 = 0; pat_idx1 < model.num_patterns; pat_idx1++) {
-		cv::Mat scaled1;
-		auto patt1 = model.patterns[pat_idx1];
-		cv::resize(patt1, scaled1, cv::Size(128, 128), 0.0, 0.0, cv::INTER_AREA);
-		std::cout << "Pattern: " << pat_idx1 << " | Pattern count: " << model.counts_[pat_idx1] << std::endl;
-		
-		for (auto overlay_idx = 0; overlay_idx < model.overlay_count; overlay_idx++) {
-			int index = pat_idx1 * model.overlay_count + overlay_idx;
-			int opposite_idx = pat_idx1 * model.overlay_count + ((overlay_idx + 2) % model.overlay_count);
-			auto valid_patterns = model.fit_table_[index];
-			auto overlay = overlays[overlay_idx];
-			auto opposite = overlays[(overlay_idx + 2) % model.overlay_count];
-			std::cout << "Valid Patterns: ";
-			for (auto pattern_2: valid_patterns)	std::cout << pattern_2 << ", ";
-			std::cout << std::endl;
-			std::cout << "Compat Table Length: " << model.compatible_neighbors_[opposite_idx] << std::endl;
-			std::cout << "Overlay: " << overlay << " | Opposite: " << opposite << std::endl;
+		for (auto pat_idx1 = 0; pat_idx1 < model.num_patterns; pat_idx1++) {
+			cv::Mat scaled1;
+			auto patt1 = model.patterns[pat_idx1];
+			cv::resize(patt1, scaled1, cv::Size(128, 128), 0.0, 0.0, cv::INTER_AREA);
+			std::cout << "Pattern: " << pat_idx1 << " | Pattern count: " << model.counts_[pat_idx1] << std::endl;
```

Comment:
Any access to a private method is not allowed from outside that object. There are a few ways to get around this: such as moving the rendering inside the model or changing the visibility of methods (with proper refactoring). I recommend just changing visibility and renaming methods to "model._{private method name}" (note that I currently have this backwards in the code, my bad).

Reply:
BTW fix this for all private accesses